src/roles/dullahan.py

```diff
@@ -5,76 +5,73 @@
-    if not victim:
+    target = get_target(var, wrapper, re.split(" +", message)[0])
+    if not target:
         return
 
-    if victim == nick:
-        pm(cli, nick, messages["no_suicide"])
+    if target is wrapper.source:
+        wrapper.pm(messages["no_suicide"])
         return
 
-    orig = victim
-    evt = Event("targeted_command", {"target": victim, "misdirection": True, "exchange": True})
-    evt.dispatch(cli, var, "kill", nick, victim, frozenset({"detrimental"}))
+    orig = target
+    evt = Event("targeted_command", {"target": target.nick, "misdirection": True, "exchange": True})
+    evt.dispatch(wrapper.client, var, "kill", wrapper.source.nick, target.nick, frozenset({"detrimental"}))
     if evt.prevent_default:
         return
-    victim = evt.data["target"]
+    target = users._get(evt.data["target"]) # FIXME: Need to fix once targeted_command uses the new API
 
-    KILLS[nick] = victim
+    KILLS[wrapper.source] = target
 
-    msg = messages["wolf_target"].format(orig)
-    pm(cli, nick, messages["player"].format(msg))
+    wrapper.pm(messages["player"].format(messages["wolf_target"].format(orig)))
 
-    debuglog("{0} ({1}) KILL: {2} ({3})".format(nick, get_role(nick), victim, get_role(victim)))
+    debuglog("{0} ({1}) KILL: {2} ({3})".format(wrapper.source, get_role(wrapper.source.nick), target, get_role(target.nick)))
 
-    chk_nightdone(cli)
+    chk_nightdone(wrapper.client)
 
-@cmd("retract", "r", chan=False, pm=True, playing=True, phases=("night",), roles=("dullahan",))
-def dullahan_retract(cli, nick, chan, rest):
+@command("retract", "r", chan=False, pm=True, playing=True, phases=("night",), roles=("dullahan",))
+def dullahan_retract(var, wrapper, message):
     """Removes a dullahan's kill selection."""
-    if nick not in KILLS:
-        return
-    if nick in KILLS:
-        del KILLS[nick]
-    pm(cli, nick, messages["retracted_kill"])
+    if KILLS.pop(wrapper.source, None):
+        wrapper.pm(messages["retracted_kill"])
 
 @event_listener("player_win")
 def on_player_win(evt, var, user, role, winner, survived):
     if role != "dullahan":
         return
-    alive = set(list_players())
-    if not TARGETS[user.nick] & alive:
+    alive = set(get_players())
+    if not TARGETS[user] & alive:
         evt.data["iwon"] = True
 
 @event_listener("del_player")
 def on_del_player(evt, cli, var, nick, nickrole, nicktpls, death_triggers):
-    for h,v in list(KILLS.items()):
-        if v == nick:
-            pm(cli, h, messages["hunter_discard"])
+    for h, v in list(KILLS.items()):
+        if v.nick == nick:
+            h.send(messages["hunter_discard"])
             del KILLS[h]
-        elif h == nick:
+        elif h.nick == nick:
             del KILLS[h]
     if death_triggers and nickrole == "dullahan":
         pl = evt.data["pl"]
-        targets = TARGETS[nick] & set(pl)
+        targets = TARGETS[users._get(nick)].union(users._get(x) for x in pl) # FIXME
```

Comment:
Why did you change this to a union? The original code is an intersection. Union is very wrong here.